packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,408 @@
+        }
+        const [stateChainError] = await Promise.all([
+          prisma.stateChainError.findUniqueOrThrow({
+            where: { id: swapRequest.ignoredEgresses.at(0)!.stateChainErrorId },
+          }),
+        ]);
+        error = {
+          name: stateChainError.name,
+          message: stateChainError.docs,
+        };
+      } else if (hasAbortedBroadcasts) {
+        failureMode = Failure.BroadcastAborted;
+      }
+    } else if (egress?.broadcast?.succeededAt) {
+      state = StateV2.Complete;
+    } else if (egress?.broadcast) {
+      const pendingBroadcast = await getPendingBroadcast(egress.broadcast);
+      if (pendingBroadcast) {
+        state = StateV2.Broadcasted;
+        egressTrackerTxRef = pendingBroadcast.tx_ref;
+      }
+    } else if (egress) {
+      state = StateV2.EgressScheduled;
+    } else if (swapRequest?.swaps.some((s) => s.swapExecutedAt)) {
+      state = StateV2.Swapping;
+    } else if (swapRequest?.depositReceivedAt) {
+      state = StateV2.DepositReceived;
+    } else {
+      state = StateV2.AwaitingDeposit;
+    }
+
+    const internalSrcAsset = readField(swapRequest, swapDepositChannel, failedSwap, 'srcAsset');
+    const internalDestAsset = readField(swapRequest, swapDepositChannel, 'destAsset');
+
+    let pendingDeposit;
+    if (
+      internalSrcAsset &&
+      state === StateV2.AwaitingDeposit &&
+      swapDepositChannel?.depositAddress
+    ) {
+      pendingDeposit = await getPendingDeposit(
+        assetConstants[internalSrcAsset].chain,
+        assetConstants[internalSrcAsset].asset,
+        swapDepositChannel.depositAddress,
+      );
+    }
+
+    let ccmParams;
+    if (readField(swapRequest, swapDepositChannel, 'ccmGasBudget')) {
+      ccmParams = {
+        gasBudget: readField(swapRequest, swapDepositChannel, 'ccmGasBudget')?.toFixed(),
+        message: readField(swapRequest, swapDepositChannel, 'ccmMessage'),
+      };
+    }
+
+    let effectiveBoostFeeBps;
+
+    if (swapDepositChannel && swapDepositChannel.maxBoostFeeBps > 0) {
+      if (swapRequest) {
+        effectiveBoostFeeBps = swapRequest.effectiveBoostFeeBps ?? undefined;
+      } else if (swapDepositChannel.failedBoosts.length > 0) {
+        effectiveBoostFeeBps = 0;
+      }
+    }
+    const { srcAsset, destAsset } = {
+      srcAsset: swapDepositChannel?.srcAsset || swapRequest?.srcAsset,
+      destAsset: swapDepositChannel?.destAsset || swapRequest?.destAsset,
+    };
+
+    const depositTransactionRef =
+      swapRequest?.depositTransactionRef ??
+      pendingDeposit?.transactionHash ??
+      failedSwap?.depositTransactionRef ??
+      undefined;
+
+    const sortedSwaps = swapRequest?.swaps.sort((a, b) => {
+      if (a.swapExecutedAt && b.swapExecutedAt) {
+        return a.swapExecutedAt.valueOf() - b.swapExecutedAt.valueOf() ? 1 : -1;
+      }
+      return a.swapExecutedAt && !b.swapExecutedAt ? 1 : 0;
+    });
+    // const fees = (swapRequest?.fees ?? []).concat(swap?.fees ?? []);
```

Comment:
maybe we can add the swap id or the swap request id to the fees to make it clear where they came from but allow us to put them all in one array still

Reply:
i'll wait for the broker fee refactor to handle fees. I think that might change how we want to structure it.

Reply:
didn't realize it was done. will do it here